# HG changeset patch
# User jwe
# Date 823922455 0
#      Sat Feb 10 03:20:55 1996 +0000
# Node ID ca0b39eccd65cd6fa01135a3feb326e9cf8d9f40
# Parent  c02829e877a18ab1a02aed7d72fae676991409a6
[project @ 1996-02-10 03:20:55 by jwe]
Initial revision

diff --git a/libcruft/misc/f77-extern.cc b/libcruft/misc/f77-extern.cc
new file mode 100644
--- /dev/null
+++ b/libcruft/misc/f77-extern.cc
@@ -0,0 +1,48 @@
+// misc-extern.cc                                     -*- C++ -*-
+/*
+
+Copyright (C) 1996 John W. Eaton
+
+This file is part of Octave.
+
+Octave is free software; you can redistribute it and/or modify it
+under the terms of the GNU General Public License as published by the
+Free Software Foundation; either version 2, or (at your option) any
+later version.
+
+Octave is distributed in the hope that it will be useful, but WITHOUT
+ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+for more details.
+
+You should have received a copy of the GNU General Public License
+along with Octave; see the file COPYING.  If not, write to the Free
+Software Foundation, 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
+
+*/
+
+#ifdef HAVE_CONFIG_H
+#include <config.h>
+#endif
+
+#include "f77-fcn.h"
+#include "lo-error.h"
+
+// This whole file is a kluge just to avoid unresolved symbol errors
+// when creating shared versions of libcruft.
+
+// So we can check to see if an exception has occurred.
+int f77_exception_encountered = 0;
+
+// For setjmp/longjmp.
+jmp_buf f77_context;
+
+// Pointer to the current error handling function.
+liboctave_error_handler current_liboctave_error_handler = liboctave_fatal;
+
+/*
+;;; Local Variables: ***
+;;; mode: C++ ***
+;;; page-delimiter: "^/\\*" ***
+;;; End: ***
+*/
